service/s3/s3manager/upload.go

```diff
@@ -399,19 +401,19 @@ func (u *uploader) init() {
 	}
 
 	// Try to get the total size for some optimizations
-	u.initSize()
+	return u.initSize()
 }
 
 // initSize tries to detect the total stream size, setting u.totalSize. If
 // the size is not known, totalSize is set to -1.
-func (u *uploader) initSize() {
+func (u *uploader) initSize() error {
 	u.totalSize = -1
 
 	switch r := u.in.Body.(type) {
```

Comment:
Would we be adding additional cases in this switch case?

Reply:
This is a special check for an `io.Seeker`, if the reader is not an `io.Seeker` then the uploader must fallback to streaming the upload to S3.